WordPress/gutenberg
# Block Serialization Default Parser: Include TypeScript type declarations

## What?


This PR adds types to the `@wordpres/block-serialization-default-parser` and exposes TypeScript types for the package.

## Why?


To increase the code quality and the documentation.

It's also part of an effort to add types to `@wordpress/blocks` package that depends on this package.

## How?


Adds JSDoc comment with types for `@wordpres/block-serialization-default-parser`. 

## Testing Instructions





All CI check should pass. Locally, `npm run build:package-types` should finish with no errors.
